.pre-commit-config.yaml

```
@@ -0,0 +1,39 @@
+fail_fast: true
+
+exclude: |
+  (?x)^(
+    .*/(snapshots)/.*|
+  )$
+
+repos:
+  - repo: https://github.com/abravalheri/validate-pyproject
+    rev: v0.16
+    hooks:
+      - id: validate-pyproject
+
+  - repo: https://github.com/crate-ci/typos
+    rev: v1.18.2
+    hooks:
+      - id: typos
+
+  - repo: local
+    hooks:
+      - id: cargo-fmt
+        name: cargo fmt
+        entry: cargo fmt --
+        language: system
+        types: [rust]
+        pass_filenames: false # This makes it a lot faster
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v3.1.0
+    hooks:
+      - id: prettier
+        types: [yaml]
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.2.1
+    hooks:
+      - id: ruff-format
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
```

```suggestion
        args: [--fix, --exit-non-zero-on-fix, --show-fixes]
```

Also, I believe I've been told `ruff-format` should follow `ruff`.

---

Also copied from Ruff, no opinions either way.